src/utils/actionUtils.ts

```diff
@@ -1,86 +1,35 @@
 import * as core from "@actions/core";
-import * as exec from "@actions/exec";
-import * as glob from "@actions/glob";
-import * as io from "@actions/io";
-import * as fs from "fs";
-import * as path from "path";
-import * as util from "util";
-import * as uuidV4 from "uuid/v4";
 
-import {
-    CacheFilename,
-    CompressionMethod,
-    Outputs,
-    RefKey,
-    State
-} from "../constants";
-import { ArtifactCacheEntry } from "../contracts";
+import { Outputs, RefKey, State } from "../constants";
 
-// From https://github.com/actions/toolkit/blob/master/packages/tool-cache/src/tool-cache.ts#L23
-export async function createTempDirectory(): Promise<string> {
-    const IS_WINDOWS = process.platform === "win32";
-
-    let tempDirectory: string = process.env["RUNNER_TEMP"] || "";
-
-    if (!tempDirectory) {
-        let baseLocation: string;
-        if (IS_WINDOWS) {
-            // On Windows use the USERPROFILE env variable
-            baseLocation = process.env["USERPROFILE"] || "C:\\";
-        } else {
-            if (process.platform === "darwin") {
-                baseLocation = "/Users";
-            } else {
-                baseLocation = "/home";
-            }
-        }
-        tempDirectory = path.join(baseLocation, "actions", "temp");
-    }
-
-    const dest = path.join(tempDirectory, uuidV4.default());
-    await io.mkdirP(dest);
-    return dest;
-}
-
-export function getArchiveFileSize(path: string): number {
-    return fs.statSync(path).size;
-}
-
-export function isExactKeyMatch(
-    key: string,
-    cacheResult?: ArtifactCacheEntry
-): boolean {
+export function isExactKeyMatch(key: string, cacheKey?: string): boolean {
     return !!(
-        cacheResult &&
-        cacheResult.cacheKey &&
-        cacheResult.cacheKey.localeCompare(key, undefined, {
+        cacheKey &&
+        cacheKey.localeCompare(key, undefined, {
             sensitivity: "accent"
         }) === 0
     );
 }
 
-export function setCacheState(state: ArtifactCacheEntry): void {
-    core.saveState(State.CacheResult, JSON.stringify(state));
+export function setCacheState(state: string): void {
+    core.saveState(State.CacheResult, state);
```

Comment:
Let's change the name of this `CacheResult` constant since we're only storing the key now in the state